internal/controller/storageclient_controller.go

```diff
@@ -273,7 +290,7 @@ func (r *storageClientReconcile) reconcileDynamicWatches() error {
 	if err := r.controller.Watch(
 		source.Kind(
 			r.cache,
-			client.Object(&groupsnapapi.VolumeGroupSnapshotContent{}),
+			client.Object(obj),
```

Comment:
```suggestion
			obj,
```
this should work since the received obj is already a `client.Object`